src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -121,6 +137,29 @@ private static bool IsEndOfLine(SyntaxTrivia trivia)
                     || trivia.IsDirective;
             }
 
+            /// <summary>
+            /// Returns the first end of line found in any of the <see cref="SyntaxTriviaList"/> lists.
+            /// </summary>
+            private static SyntaxTrivia? GetEndOfLine(params SyntaxTriviaList[] lists)
+            {
+                foreach (var list in lists)
+                {
+                    foreach (var trivia in list)
+                    {
+                        if (trivia.Kind() == SyntaxKind.EndOfLineTrivia)
+                        {
+                            return trivia;
+                        }
+                        if (trivia.IsDirective && trivia.GetStructure() is DirectiveTriviaSyntax directiveTrivia)
+                        {
+                            return directiveTrivia.EndOfDirectiveToken.TrailingTrivia
+                                .FirstOrDefault(t => t.Kind() == SyntaxKind.EndOfLineTrivia);
```

Comment:
1. shouldn't this only return *if* the trailing trivia found an end-of-line?  otherwise, you're skipping future trivia that may have an end-of-line.   
2. do we have tests fort hsi case?  i'm trying to figure out when it's relevant.
3. are there other cases of structure trivia that also have endoflines we'd care about (like doc comments)?

Reply:
1. Yes you are right, I've corrected that.
2. Yes, we have two tests that go through this code [TestRemove_KeepDirectives](https://github.com/dotnet/roslyn/pull/23165/commits/dc5568b3f3a3dfb4a67ced53f410d29c8dd5ae69#diff-0e0e5cee589dd5835bc357a2c0406c46R2409) and [TestRemove_KeepUnbalancedDirectives](https://github.com/dotnet/roslyn/pull/23165/commits/dc5568b3f3a3dfb4a67ced53f410d29c8dd5ae69#diff-0e0e5cee589dd5835bc357a2c0406c46R2358)
3. No, because this method is called only when [HasEndOfLine](https://github.com/dotnet/roslyn/blob/5e37e3d5383b44cb9863e85373434a07364a087b/src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs#L163) returns true, which does not include doc comments.